src/raw/unix.rs

```diff
@@ -414,6 +418,32 @@ impl Socket {
     }
 
 
+    ///Sets whether this socket will be inherited to child porcesses or not.
+    pub fn set_inheritable(&self, value: bool) -> io::Result<()> {
+        // Some (or possibly all?) OS's support the `FIOCLEX` and `FIONCLEX`

```

Comment:
I believe one strategy to check for availability of these flags would to try to set it ONCE, but it would make your code overly complex so it is up to you